### ISSUE TITLE
After upgrading 28.0.0 bz2 module missing 

### ISSUE THREAD

**Issue by jacme31** [nextcloud/docker#2118]
After upgrading nextcloud 28.0.0  (nextcloud:latest on dockerhub)  I am getting the following error :
```
There are some warnings regarding your setup.
This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2
```

Otherwise, apart from a few apps that don't work , I haven't identified any major problems.

**Comment by waldner** [comment]
Furthermore, if you click on the documentation link (`For more details see the [documentation ↗`) it takes you to a non-existing page: https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules

**Comment by wolrechris** [comment]
> Furthermore, if you click on the documentation link (`For more details see the [documentation ↗]`) it takes you to a non-existing page: https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules

True, but this is most likely an issue with Nextcloud in general / the fact that they have not yet updated the documentation and not the nextcloud docker container in particular.

**Comment by molhamalnasr** [comment]
yes i confirm it. after i updated my nextcloud app to version 28.0.0 i got the same warning about the bz2. and also the link is not working. this is the correct link to the recommended modules from Nextcloud documentation: https://docs.nextcloud.com/server/stable/admin_manual/installation/source_installation.html

**Comment by tzerber** [comment]
I can confirm i have the same issue on a full fresh install of latest fpm docker image.
 
As a workaround i have created a local Dockerfile that uses NC as base image and just installs the bz2 php module, to keep it across downs/ups of compose, since it was fresh install and it needed a bit of fiddling with.

```
FROM nextcloud:fpm
RUN apt-get update && apt-get install -y libbz2-dev && rm -rf /var/lib/apt/lists/*
RUN docker-php-ext-install bz2
```

**Comment by dugite-code** [comment]
Old pull request regarding this issue: https://github.com/nextcloud/docker/pull/1351

**Comment by linucksrox** [comment]
Just to hammer the point that was made in that other issue, I agree that Nextcloud can't include everything in the image. But now they "highly recommend" bz2 and there's a big warning about it on the Security & Setup Warnings page:
> This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).

That means on a brand new installation using Docker, users are expected to figure out how to make their own customized version of a Dockerfile just to add this dependency. It doesn't just work out of the box. Sorry, but that is dumb.

**Comment by mortee** [comment]
+1

**Comment by sparklyballs** [comment]
> Just to hammer the point that was made in that other issue, I agree that Nextcloud can't include everything in the image. But now they "highly recommend" bz2 and there's a big warning about it on the Security & Setup Warnings page:
> 
> > This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).
> 
> That means on a brand new installation using Docker, users are expected to figure out how to make their own customized version of a Dockerfile just to add this dependency. It doesn't just work out of the box. Sorry, but that is dumb.

certain sense of deja-vu here too....

https://github.com/nextcloud/docker/issues/1414

**Comment by mortee** [comment]
> #1414

That one apparently took a quick 1,5 years to resolve, fingers crossed for this new one 😃

**Comment by linucksrox** [comment]
@J0WI in that case it's strange that we are seeing this new warning that wasn't there before. It sounds like the bug isn't the missing module, but the actual message that needs to be removed.

**Comment by jjasoncool** [comment]
@J0WI I just fix it.
If there is anything need help, please let me know.
I'm glad to help this.

**Comment by cobran89** [comment]
how to fix this whe I use Podman, not Docker. Thanks in advance/

**Comment by joshtrichards** [comment]
My suggestion is that no one build a custom image merely to fix this warning that isn't impacting anything operationally. The check for bz2 was added in a bundle of other php module checks. There's a 50/50 chance bz2 gets removed from the docs and the check goes away.

Tracking upstream in nextcloud/server#42342

**Comment by heikkiheer** [comment]
> My suggestion is that no one build a custom image merely to fix this warning that isn't impacting anything operationally. The check for bz2 was added in a bundle of other php module checks. There's a 50/50 chance bz2 gets removed from the docs and the check goes away.
> 
> Tracking upstream in [nextcloud/server#42342](https://github.com/nextcloud/server/issues/42342)

I agree strongly.

**Comment by Staubgeborener** [comment]
> I can confirm i have the same issue on a full fresh install of latest fpm docker image.
> 
> As a workaround i have created a local Dockerfile that uses NC as base image and just installs the bz2 php module, to keep it across downs/ups of compose, since it was fresh install and it needed a bit of fiddling with.
> 
> ```
> FROM nextcloud:fpm
> RUN apt-get update && apt-get install -y libbz2-dev && rm -rf /var/lib/apt/lists/*
> RUN docker-php-ext-install bz2
> ```

Could you create a gist and explain how you did that? I create a Dockerfile with this contest (just used `FROM nextcloud:latest`) and add
```shell
    build:
      context: .
      dockerfile: Dockerfile
```

to my `docker-compose.yml`. But this doesn't install `bz2`. The web interface tells me, that `bz2` is still missing.

So a minimal example of a `docker-compose.yml` and `Dockerfile` is appreciated. Do you run `docker-compose up -d` or `docker-compose build`?

**Comment by tzerber** [comment]
> > I can confirm i have the same issue on a full fresh install of latest fpm docker image.
> > As a workaround i have created a local Dockerfile that uses NC as base image and just installs the bz2 php module, to keep it across downs/ups of compose, since it was fresh install and it needed a bit of fiddling with.
> > ```
> > FROM nextcloud:fpm
> > RUN apt-get update && apt-get install -y libbz2-dev && rm -rf /var/lib/apt/lists/*
> > RUN docker-php-ext-install bz2
> > ```
> 
> Could you create a gist and explain how you did that? I create a Dockerfile with this contest (just used `FROM nextcloud:latest`) and add
> 
> ```shell
>     build:
>       context: .
>       dockerfile: Dockerfile
> ```
> 
> to my `docker-compose.yml`. But this doesn't install `bz2`. The web interface tells me, that `bz2` is still missing.
> 
> So a minimal example of a `docker-compose.yml` and `Dockerfile` is appreciated. Do you run `docker-compose up -d` or `docker-compose build`?

Hello,

Create an empty folder named app in the location of the compose file. Inside create the following dockerfile, and edit as needed, in my case i need ffmpeg for video files preview as well, but feel free to remove it.
```dockerfile
FROM nextcloud:28.0.1-fpm

RUN apt-get update && apt-get install -y libbz2-dev ffmpeg && rm -rf /var/lib/apt/lists/*
RUN docker-php-ext-install bz2
```

Then my compose file looks like this (keep in mind, it's not the default, but you can adjust it as needed)
```yaml
  app:
    build: ./app
    restart: unless-stopped
    logging:
      driver: "json-file"
      options:
        max-size: "50m"
    volumes:
      - nextcloud:/var/www/html
      - ./opcache-recommed.ini:/usr/local/etc/php/conf.d/opcache-recommended.ini
    environment:
      - POSTGRES_HOST=db
      - REDIS_HOST=redis
      - PHP_MEMORY_LIMIT=1024M
    env_file:
      - db.env
    depends_on:
      - db
      - redis
  cron:
    build: ./app
    restart: unless-stopped
    logging:
      driver: "json-file"
      options:
        max-size: "50m"
    volumes:
      - nextcloud:/var/www/html
    entrypoint: /cron.sh
    depends_on:
      - db
      - redis
```
Then start the containers with the build and pull params (note, you can also do a separate compose build --pull, but i find this way more "safe")

```bash
docker compose up --build --pull always -d
```

Few notes: 
1) If your instance is production one (or it's somewhat important to you) you should avoid using latest tags. Keep the version, and when new version is released - test it on separate instance then upgrade your main one.

2) In my example i use `restart: unless-stopped`. People will argue about that, but when you have a container that is causing problems you should not use `restart: always`, because it can have some not-so-good side effects.

3) Using the same approach i have a custom Nginx container built because the inconsistencies between Apache, Nginx , Chrome, Firefox and their recent versions when it comes to mjs files.

**Comment by Staubgeborener** [comment]
Thanks, this is working @tzerber.

> Then start the containers with the build and pull params (note, you can also do a separate compose build --pull, but i find this way more "safe")
> 
> ```shell
> docker compose up --build --pull always -d
> ```
I get a `unknown flag: --pull` error, but `docker-compose pull && docker-compose up --build -d` is working for me. 

>     1. If your instance is production one (or it's somewhat important to you) you should avoid using latest tags. Keep the version, and when new version is released - test it on separate instance then upgrade your main one.

I wrote myself some sort of [update script](https://github.com/Staubgeborener/docker-backups/blob/main/nextcloud/update-nextcloud.sh) so i can pass the version like `./update-nextcloud.sh 28` or `./update-nextcloud.sh latest`. The parameter is passed as a variable to both `docker-compose.yml` and the `Dockerfile`.

>     3. Using the same approach i have a custom Nginx container built because the inconsistencies between Apache, Nginx , Chrome, Firefox and their recent versions when it comes to mjs files.

I also want to switch from apache to nginx, but it is difficult to realize this with the appropriate containers. I could use `nextcloud:*-fpm` release, but It's quite hard to link this to nginx because of lack of time (do you have any more information?).

**Comment by tzerber** [comment]
> Thanks, this is working @tzerber.
> 
> > Then start the containers with the build and pull params (note, you can also do a separate compose build --pull, but i find this way more "safe")
> > ```shell
> > docker compose up --build --pull always -d
> > ```
>
> I get a `unknown flag: --pull` error, but `docker-compose pull && docker-compose up --build -d` is working for me.
>

This is because you are using `docker-compose` instead of `docker compose`. Notice the difference in the command executed. You can visit docker documentation about why and what is like this. Same goes for my compose, your file is probably called docker-compose.yml where mine is just compose.yml and it has not `version: X` line in it. 

> > ```
> > 1. If your instance is production one (or it's somewhat important to you) you should avoid using latest tags. Keep the version, and when new version is released - test it on separate instance then upgrade your main one.
> > ```
> 
> I wrote myself some sort of [update script](https://github.com/Staubgeborener/docker-backups/blob/main/nextcloud/update-nextcloud.sh) so i can pass the version like `./update-nextcloud.sh 28` or `./update-nextcloud.sh latest`. The parameter is passed as a variable to both `docker-compose.yml` and the `Dockerfile`.
> 
> > ```
> > 3. Using the same approach i have a custom Nginx container built because the inconsistencies between Apache, Nginx , Chrome, Firefox and their recent versions when it comes to mjs files.
> > ```
> 
> I also want to switch from apache to nginx, but it is difficult to realize this with the appropriate containers. I could use `nextcloud:*-fpm` release, but It's quite hard to link this to nginx because of lack of time (do you have any more information?).

I recently updated some of the examples [in the docker examples section](https://github.com/nextcloud/docker/tree/master/.examples/docker-compose/with-nginx-proxy/postgres/fpm) you can use that as example. If you can't manage it, i can probably write you a proper example, but we're derailing this issue thread sooo much that i expect someone from maintainers will come yell at us :D

**Comment by witzker** [comment]
Same warning with NC 18.0.1 on Debian.
Operating System: Linux 5.10.0-25-amd64 x86_64
How to solve this?

**Comment by tzerber** [comment]
> Same warning with NC 18.0.1 on Debian. Operating System: Linux 5.10.0-25-amd64 x86_64 How to solve this?

This is the docker section. In general, you need to install the php bz2 extension depending on your PHP version, probably php8.2-bz2 to get rid of the warning.

**Comment by XHyperDEVX** [comment]
> This is the docker section. In general, you need to install the php bz2 extension depending on your PHP version, probably php8.2-bz2 to get rid of the warning.

How do I do that? `apt install php8.2-bz2` does not work, has no installation candidate. I am using the latest stable image.

**Comment by witzker** [comment]
Many THX for quick answer.
Sorry, I'm new to all this server stuff. Could you Pls. be so kind and tell me the command for that?
To get rid of it quickly!
Many THX

**Comment by mortee** [comment]
For me, this Dockerfile does the trick:
```
FROM nextcloud:28
RUN apt-get -y update && \
  apt-get -yqq --no-install-recommends install libbz2-dev && \
  rm -rf /var/lib/apt/lists/*
RUN docker-php-ext-install bz2
```

**Comment by tzerber** [comment]
> Many THX for quick answer. Sorry, I'm new to all this server stuff. Could you Pls. be so kind and tell me the command for that? To get rid of it quickly! Many THX



> > This is the docker section. In general, you need to install the php bz2 extension depending on your PHP version, probably php8.2-bz2 to get rid of the warning.
> 
> How do I do that? `apt install php8.2-bz2` does not work, has no installation candidate. I am using the latest stable image.

Both of you : again, this is the docker section, not the debian one.  I can probably help both, but not here, because this issue is not a discussion group and it's not the proper way of getting things done.  Open a proper ticket in this section, https://github.com/nextcloud/server/issues and provide a proper information on your setup including the version of php that you're using.

**Comment by XHyperDEVX** [comment]
I and I think everyone else here would prefer this package to simply be added to the image anyway, why not just do that? where is the problem?

**Comment by witzker** [comment]
> Both of you : again, this is the docker section, not the debian one
Sorry, I'm new & THX for your advice

**Comment by tzerber** [comment]
> I and I think everyone else here would prefer this package to simply be added to the image anyway, why not just do that? where is the problem?

Because it's not required/not used anywhere. It's the check that needs updating. Unless there's an app that need bz2 you generally don't need it. The downside is that you get a warning that it's missing, but no actual functionality depends on it.

**Comment by XHyperDEVX** [comment]
Okay, that makes sense! and what is the problem with removing the warning when it is not needed? i mean the problem has existed for a long time

**Comment by tzerber** [comment]
> Okay, that makes sense! and what is the problem with removing the warning when it is not needed? i mean the problem has existed for a long time

You can ask that in the issues section of Nextcloud itself. This is the docker section, and people here can't really change it.
For future reference : https://github.com/nextcloud/server/issues/42342 .

**Comment by Banditen01** [comment]
> > I and I think everyone else here would prefer this package to simply be added to the image anyway, why not just do that? where is the problem?
> 
> Because it's not required/not used anywhere. It's the check that needs updating. Unless there's an app that need bz2 you generally don't need it. The downside is that you get a warning that it's missing, but no actual functionality depends on it.

At least for me I get all different errors/warnings in logfile without bz2 (NC 28.0.1), with bz2 I get a clean logfile and get rid of the warning = green light. Some kind of trick it must be doing!?

**Comment by tzerber** [comment]
> > > I and I think everyone else here would prefer this package to simply be added to the image anyway, why not just do that? where is the problem?
> > 
> > 
> > Because it's not required/not used anywhere. It's the check that needs updating. Unless there's an app that need bz2 you generally don't need it. The downside is that you get a warning that it's missing, but no actual functionality depends on it.
> 
> At least for me I get all different errors/warnings in logfile without bz2 (NC 28.0.1), with bz2 I get a clean logfile and get rid of the warning = green light. Some kind of trick it must be doing!?
Can you provide the log with the errors you get from the lack of bz2? Even if it's not directly bz2 related it will be quite helpful.

**Comment by Banditen01** [comment]
> > > > I and I think everyone else here would prefer this package to simply be added to the image anyway, why not just do that? where is the problem?
> > > 
> > > 
> > > Because it's not required/not used anywhere. It's the check that needs updating. Unless there's an app that need bz2 you generally don't need it. The downside is that you get a warning that it's missing, but no actual functionality depends on it.
> > 
> > 
> > At least for me I get all different errors/warnings in logfile without bz2 (NC 28.0.1), with bz2 I get a clean logfile and get rid of the warning = green light. Some kind of trick it must be doing!?
> 
> Can you provide the log with the errors you get from the lack of bz2? Even if it's not directly bz2 related it will be quite helpful.

@tzerber have to recall my statement, had some struggle to prune and pull, got already exists when I tried to go back to NC latest. After managed to clean up I can't replicate logfile issues, I only get bz2 missing warning message.

**Comment by Ramalama2** [comment]
Oh boys, the docker images for 28.0.1 still miss the bz2 package.

We had an similar issue already with another package that was missing i dont remember if it was v24 or v26, however, it took an eternity till the pull reached the public images.
Something has to change, because this will happen every time nextcloud needs something new as package again next time...

Cheers

**Comment by joshtrichards** [comment]
@Ramalama2 That's not what is going on here. There is an upstream issue in Server this one tied into. This one is blocked on clarification there. The `bz2` is not a new requirement - but a new check was added for it in Server. No one has complained about the bz2 module not being present in the image until the check was added. :) This entire matter has zero impact on functionality.

**Comment by robertoschwald** [comment]
Thats not the point. The point is that a docker image is published which causes warning messages. All manual things done to fix the warning message are workarounds. For us this is a no-go as Nextcloud runs in K8S. We neither want to use our own image, nor want to manually change running pods. 
All we ask is to simply install bz2 in the official Dockerfile until it is clarified in the Server issue what to do in the future.

**Comment by tzerber** [comment]
> Thats not the point. The point is that a docker image is published which causes warning messages. All manual things done to fix the warning message are workarounds. For us this is a no-go as Nextcloud runs in K8S. We neither want to use our own image, nor want to manually change running pods. All we ask is to simply install bz2 in the official Dockerfile until it is clarified in the Server issue [nextcloud/server#42342](https://github.com/nextcloud/server/issues/42342) what to do in the future.

I don't agree here. The whole idea of using docker is to use only the software that's needed for an application to function properly, nothing more nothing less. The presence of bz2 php extension in the container is not needed, no app or functionality is requiring it. You will get surprised how much software on this planet works fine despite throwing a lot more core warnings (and more obvios ones) than this single one we're discussing here. 

 [The official docker image of Nextcloud](https://github.com/nextcloud/all-in-one) also does not have that php extension.

**Comment by thomasmerz** [comment]
What about https://github.com/nextcloud/docker? This seems also to be "official", therefore I'm using it since some years and I find also this warning in https://mynextcloud.mydomain.tld/settings/admin/overview:

> This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2

🤷🏼‍♂️

**Comment by tzerber** [comment]
> What about https://github.com/nextcloud/docker? This seems also to be "official", therefore I'm using it since some years and I find also this warning in https://mynextcloud.mydomain.tld/settings/admin/overview:
> 

This is more of a contribution-powered project, and it's not related to Nextcloud GmbH, just a bunch of folks (nextcloud-employed or not) trying to offer micro-service variant of the official Nextcloud / Nextcloud-Docker-AIO variant. And the folks here are trying their best to keep it as 1:1 with the upstream as possible, and if something is wrong or causing issues coming from upstream to report/fix them there, not just turn this repository in a Nextcloud-workarounds one. 


> > This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2
> 
> 🤷🏼‍♂️

While the problem with the warning is settled with upstream Nextcloud (should or we shouldn't install bz2 module regardless of docker environment or not ) you can workaround this problem if you want to get rid of the warning by using a modified Docker file as shown here : 
https://github.com/nextcloud/docker/issues/2118#issuecomment-1854460093 
Keep in mind that this warning is just a warning, nothing in Nextcloud depends on bz2. Since this is a open source project, you can even use that Dockerfile modification, edit it if/as needed, make a container and push it to repository like docker hub and then you can use it in your compose / k8s / other use-cases without the need of building it every time and so on.

**Comment by Ramalama2** [comment]
> @Ramalama2 That's not what is going on here. There is an upstream issue in Server this one tied into. This one is blocked on clarification there. The `bz2` is not a new requirement - but a new check was added for it in Server. No one has complained about the bz2 module not being present in the image until the check was added. :) This entire matter has zero impact on functionality.

Thanks for Clarification!
If nothing uses it, then there is indeed no need to add that Package, just this stupid warning message is annoying.



> > Thats not the point. The point is that a docker image is published which causes warning messages. All manual things done to fix the warning message are workarounds. For us this is a no-go as Nextcloud runs in K8S. We neither want to use our own image, nor want to manually change running pods. All we ask is to simply install bz2 in the official Dockerfile until it is clarified in the Server issue [nextcloud/server#42342](https://github.com/nextcloud/server/issues/42342) what to do in the future.
> 
> I don't agree here. The whole idea of using docker is to use only the software that's needed for an application to function properly, nothing more nothing less. The presence of bz2 php extension in the container is not needed, no app or functionality is requiring it. You will get surprised how much software on this planet works fine despite throwing a lot more core warnings (and more obvios ones) than this single one we're discussing here.
> 
> [The official docker image of Nextcloud](https://github.com/nextcloud/all-in-one) also does not have that php extension.

This is the Stupidest thing ever, i though im using the Official Docker image already :-)
Ill use this images everywhere, i think i have like 5 instances at different locations xD

The Stupid thing that i mean is:
(github domain)/nextcloud/docker
(github domain)/nextcloud/all-in-one

This makes no sense that /docker is community driven and /all-in-one is official...
If someone (like me) doesn't read, or look especially for it, he would assume that both are official.

Community driven should be something like:
(github domain)/nextcloud/docker-community

However, its how it is and not really a big deal either, thanks for your replies!

Cheers

**Comment by nfp0** [comment]
Quite honestly I was also under the impression this /nextcloud/docker repository was official.
If not, why is it listed under the Nextcloud organization?

**Comment by linucksrox** [comment]
As far as I'm concerned this is official because it's part of the official nextcloud source code. Who maintains it isn't relevant, as it's literally just the source baked into a docker image, along with a migration script.

Unless nextcloud tells us otherwise, how could we say this is not official?

Also just a TLDR so far: they mistakenly added a warning in the server source that hasn't been fixed yet. For now we should be aware but safely ignore this warning and stop trying to fix it.

**Comment by rusty-snake** [comment]
> Unless nextcloud tells us otherwise, how could we say this is not official?

README.md:

> This Docker micro-service image is developed and maintained by the Nextcloud community. Nextcloud GmbH does not offer support for this Docker image.

https://hub.docker.com/_/nextcloud/:

> ⚠ This image is not officially supported by Nextcloud GmbH, use at your own risk.

---

>  If someone doesn't read, he would assume that both are official.

I understand how you came to this. However if you assume an docker image being official based on the name rather a trusted and authorized source saying so will never work. What if I add a `nextcloud-server/docker` image? Neither it works to look at starts or similar. Either have a trusted source or audit the code.

**Comment by linucksrox** [comment]
Thanks, looks like I failed to read that far down the page. At the same time, at the top it says "Docker official image" with that green badge.

And to your point about creating a similarly named image, that is not the case here. This is part of the official nextcloud "umbrella" source, meaning the company themselves added the image.

Community maintained is one thing, meaning no "official" support, but that should not be confused with "trusted". My argument is that I should be able to trust all source code under the nextcloud project whether they directly maintain it or not.

**Comment by jiriks74** [comment]
> Just to hammer the point that was made in that other issue, I agree that Nextcloud can't include everything in the image. But now they "highly recommend" bz2 and there's a big warning about it on the Security & Setup Warnings page:
> 
> > This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).
> 
> That means on a brand new installation using Docker, users are expected to figure out how to make their own customized version of a Dockerfile just to add this dependency. It doesn't just work out of the box. Sorry, but that is dumb.

I'm currently maintaining an image that solves these issues that appear when using a brand new docker image. I've installed `libmagickcore-dev` to fix `Module php-imagick in this instance has no SVG support` (I also added some things to make `NCDownloader` work). I'll be now building a new image that will fix this issue.

I'm building it for `amd64` and `arm64` so if you don't like waiting for your image to build (on RPi4 for example) you can [use my image](https://hub.docker.com/r/jiriks74/nextcloud-extended).

**Comment by linucksrox** [comment]
@jiriks74 Thanks for that. However, besides this docker image being "unofficial" whatever that means at this point, @J0WI  marked https://github.com/nextcloud/docker/pull/1351 as "wontfix" so we can't even fix the issue with a PR in this repo.

My point is that the Nextcloud team isn't allowing the community to fix the "most upstream" Docker image for lack of a better term, and using something even less "official" isn't the best long term solution.

It looks like they are pushing people toward the AIO instance, which does include bz2 by the way, but migrating to that is more complicated than using this image with a supported database, etc.

I just don't understand why this repo exists... if Nextcloud is going to be obnoxious about it and say it's not supported, but also not let the community update it as we see fit, then let's just delete it, right?

**Comment by felixauringer** [comment]
The upstream issue https://github.com/nextcloud/server/issues/42342 was just closed via https://github.com/nextcloud/server/pull/42821 by removing `bz2` from the recommended modules (thanks @J0WI). After a new release by nextcloud/server and a following release by nextcloud docker, the warning should disappear :tada:

**Comment by joshtrichards** [comment]
Quick update: The backport to v28 was merged in nextcloud/server#43013 a few moments ago. So it's officially in, but it'll be in v28.0.3. It was not able to be merged into the just released v28.0.2 because it was already in a code freeze at the time.

**Comment by abjoseph** [comment]
@joshtrichards Do you know when the latest point releases (28.0.3, 27.1.7) will have their respectively docker images published? Rough estimate or timeline?

**Comment by jacme31** [comment]
> After upgrading nextcloud 28.0.0 (nextcloud:latest on dockerhub) I am getting the following error :
> 
> ```
> There are some warnings regarding your setup.
> This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2

After upgrading to nextcloud 28.0.3 (nextcloud:latest on dockerhub), the error actually disappears.
Thanks to all contributors.
